flow/Trace.cpp

```diff
@@ -780,14 +785,22 @@ void openTraceFile(const NetworkAddress& na,
 	if (baseOfBase.empty())
 		baseOfBase = "trace";
 
-	std::string ip = na.ip.toString();
-	std::replace(ip.begin(), ip.end(), ':', '_'); // For IPv6, Windows doesn't accept ':' in filenames.
 	std::string baseName;
-	if (identifier.size() > 0) {
-		baseName = format("%s.%s.%s", baseOfBase.c_str(), ip.c_str(), identifier.c_str());
+	if (na.present()) {
+		std::string ip = na.get().ip.toString();
+		std::replace(ip.begin(), ip.end(), ':', '_'); // For IPv6, Windows doesn't accept ':' in filenames.
+
+		if (!identifier.empty()) {
+			baseName = format("%s.%s.%s", baseOfBase.c_str(), ip.c_str(), identifier.c_str());
+		} else {
+			baseName = format("%s.%s.%d", baseOfBase.c_str(), ip.c_str(), na.get().port);
+		}
+	} else if (!identifier.empty()) {
+		baseName = format("%s.%s", baseOfBase.c_str(), identifier.c_str());
 	} else {
-		baseName = format("%s.%s.%d", baseOfBase.c_str(), ip.c_str(), na.port);
+		baseName = baseOfBase;
```

Comment:
Do we need some sort of disambiguating field for when we have multiple processes? The choice of name impacts which trace files a process will clean up as it runs.

Reply:
The current code we have not calls where both identifier & na is not set, so it never goes to this else part. In the cases where the IP is not available, we pass PID as identifier.

Reply:
If we are relying on that fact, should we assert it?